src/test/javascript/portal/cart/DownloadConfirmationWindowSpec.js

```diff
@@ -9,63 +9,63 @@ describe("Portal.cart.DownloadConfirmationWindow", function() {
     var confirmationWindow;
     var downloadUrl = 'the download url';
     var downloadFilename = "imos:argo profiles";
+    var superclass;
 
     beforeEach(function() {
 
         confirmationWindow = new Portal.cart.DownloadConfirmationWindow();
+        superclass = Portal.cart.DownloadConfirmationWindow.superclass;
 
-        spyOn(confirmationWindow, 'show');
-        spyOn(confirmationWindow, 'hide');
+        spyOn(superclass, "show");
+        spyOn(superclass, "hide");
     });
 
     describe('shouldDownload', function() {
     });
 
-    describe('showIfNeeded', function() {
+    describe('show', function() {
 
         beforeEach(function() {
 
-            confirmationWindow.showIfNeeded(downloadUrl, downloadFilename);
+            confirmationWindow.show(downloadUrl, downloadFilename);
         });
 
         it('shows window if not shown', function() {
```

Comment:
How about changing all of those tests to just one? Something alone the lines of "shows window". Because we have to show it EVERY time.